lib/foreman/http_proxy/excon_connection_extension.rb

```diff
@@ -0,0 +1,13 @@
+Excon::Connection.class_eval do
+  include Foreman::HTTPProxy
+  alias_method :orig_request, :request
+
+  def request(params, &block)
+    if proxy_http_request?(@data[:proxy], @data[:host], @data[:schema])
+      log_proxied_request(http_proxy, @data[:host])
+      @data[:proxy] = http_proxy
+      setup_proxy
```

Comment:
Since the proxy can be misconfigured, it raises an Excon error (see trace below), could we somehow better rescue from this state so Foreman does not response with 500? or possibly do some validation check when the setting is saved.

```
| Excon::Error::ProxyParse (Proxy is invalid):
 |   lib/foreman/http_proxy/excon_connection_extension.rb:9:in `request'
 |   app/models/compute_resources/foreman/model/ec2.rb:65:in `regions'
 |   app/models/compute_resources/foreman/model/ec2.rb:74:in `test_connection'
 |   app/controllers/compute_resources_controller.rb:110:in `test_connection'
 |   app/controllers/concerns/application_shared.rb:15:in `set_timezone'
 |   app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
 |   app/controllers/concerns/foreman/controller/topbar_sweeper.rb:12:in `set_topbar_sweeper_controller'
 |   /home/ares/Projekty/Zdrojaky/katello/lib/katello/params_parser_wrapper.rb:12:in `call'
 |   lib/middleware/catch_json_parse_errors.rb:8:in `call'
 |   lib/middleware/tagged_logging.rb:18:in `call'
```